### PR TITLE
[Samples] Updated EF sample project to NET 6 & V4

### PR DESCRIPTION
**Pull request by kshyju**
We updated sample projects to V4 and NET6 in #786 , but left the EF sample project. Updating that as well in this PR.

Also renamed the project to be more generic (**_net-5-entity-framework.csproj_** to **_EntityFramework.csproj_**)

### CODE REVIEW

File: samples/EntityFramework/EntityFramework.csproj [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http" Version="3.0.12" />
-    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.3.0" OutputItemType="Analyzer" />
-    <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.0.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="5.0.4" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="5.0.4">
+    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http" Version="3.0.13" />
+    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.7.0" />
+    <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.8.0" />
```
Comment (liliankasem) [inline samples/EntityFramework/EntityFramework.csproj:11]: Do we want to bump this to 1.9.0 so we are on the latest

File: samples/EntityFramework/EntityFramework.csproj [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
-    <RootNamespace>net_5_entity_framework</RootNamespace>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http" Version="3.0.12" />
-    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.3.0" OutputItemType="Analyzer" />
-    <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.0.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="5.0.4" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="5.0.4">
+    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http" Version="3.0.13" />
+    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.7.0" />
```
Comment (liliankasem) [inline samples/EntityFramework/EntityFramework.csproj:10]: Do we not want to keep `OutputItemType="Analyzer"` here?

**Review by liliankasem — APPROVED** [review]
lgtm - some comments

**Review by satvu — APPROVED** [review]
lgtm!

**Pull request merged**